File: hinotsume-pass.py

import pandas as pd
east_traffic = pd.read_csv("east.csv").values
west_traffic = pd.read_csv("west.csv").values
east_pass= pd.read_csv("east2.csv").values
west_pass= pd.read_csv("west2.csv").values
east_length= pd.read_csv("el.csv").values
west_length= pd.read_csv("wl.csv").values

wl0= pd.read_csv("wl0.txt").values
wl1= pd.read_csv("wl1.txt").values
ws0= pd.read_csv("ws0.txt").values
ws1= pd.read_csv("ws1.txt").values
el0= pd.read_csv("el0.txt").values
el1= pd.read_csv("el1.txt").values
es0= pd.read_csv("es0.txt").values
es1= pd.read_csv("es1.txt").values

tt=np.linspace(0, 1800, 360000)

----



ts= 0
te= 1800


total_txt_out = 'Hinotsume Bridge, east side: 2021-12-02_12.30.54';

# Speeds of Detected Vehicles
fig = plt.figure(figsize=(12,4), dpi=300)
plt.xlim(0, 1800)
plt.plot(east_pass[:,0], east_pass[:,1], linewidth=1,label='left-going');
plt.plot(east_pass[:,0], east_pass[:,3], linewidth=1,label='right-going');
plt.xlabel('Time [s]')
plt.ylabel('Speed [km/h]')
plt.legend(fontsize=8,loc='upper right' );
plt.title('Speeds of Detected Vehicles',loc='right',fontsize=14)
fig.text(0.13, 0.9, total_txt_out);
#plt.show();
plt.savefig("east-speed.png")
plt.close()

# Speeds of Detected Vehicles
fig = plt.figure(figsize=(12,4), dpi=300)
plt.xlim(0, 1800)
plt.plot(east_pass[:,0], east_pass[:,2], linewidth=1,label='left-going');
plt.plot(east_pass[:,0], east_pass[:,4], linewidth=1,label='right-going');
plt.xlabel('Time [s]')
plt.ylabel('Length [m]')
plt.legend(fontsize=8,loc='upper right' );
plt.title('Speeds of Detected Vehicles',loc='right',fontsize=14)
fig.text(0.13, 0.9, total_txt_out);
#plt.show();
plt.savefig("east-length.png")
plt.close()

---------

west_l0= wl0;
west_l1= wl1;
west_s0= ws0;
west_s1= ws1;

total_txt_out = 'Hinotsume Bridge, west side: 2021-12-02_12.30.54';

# Car Length Histogram
fig = plt.figure(figsize=(12,4), dpi=300)
dl_bin= 0.5; # 20cm pitch
l_max0 = np.max(west_l0.flatten());
l_max1 = np.max(west_l1.flatten());
l_max = np.max((l_max1,l_max0));
b_num=int(np.floor(l_max/dl_bin));
le=b_num*dl_bin;
x1=0;
x2=le;
plt.xlim(x1, x2)
#txt_g_label='markers displacement';
plt.hist([west_l0.flatten(), west_l1.flatten()],range=(0,le),bins=b_num,rwidth=0.7,label=['left-going','right-going'])
plt.xlabel('Length [m]')
txt_ylabel='Number';
plt.ylabel(txt_ylabel)
plt.legend(fontsize=8,loc='upper right' );
plt.title('Vehicle Length Histogram (per 50 cm)',loc='right',fontsize=14)
fig.text(0.13, 0.9, total_txt_out);
plt.savefig("west-lengthhist.png")
plt.close()

# Car Speed Histogram
fig = plt.figure(figsize=(12,4), dpi=300)
dl_bin= 5; # 20cm pitch
l_max0 = np.max(west_s0.flatten());
l_max1 = np.max(west_s1.flatten());
l_max = np.max((l_max1,l_max0));
b_num=int(np.floor(l_max/dl_bin));
le=b_num*dl_bin;
x1=0;
x2=le;
plt.xlim(x1, x2)
plt.hist([west_s0.flatten(), west_s1.flatten()],range=(0,le),bins=b_num,rwidth=0.7,label=['left-going','right-going'])
plt.xlabel('Speed [km/h]')
txt_ylabel='Number';
plt.ylabel(txt_ylabel)
plt.legend(fontsize=8,loc='upper right' );
plt.title('Vehicle Speed Histogram (per 5 km/h)',loc='right',fontsize=14)
fig.text(0.13, 0.9, total_txt_out);
plt.savefig("west-speedhist.png")
plt.close()

File: hinotsume-pass.m
clear('pass_length');
clear('pass_speed');
pass_length_east(360000,2)=0;
pass_speed_east(360000,1),2)=0;

for i=1:size(trafficeast,1)
cur= trafficeast(i, fstart);
  for j=1:trafficeast(i,fend)
    if ( trafficeast(i,direction) ==1 )
      pass_length_east(cur+j-1, 1) = trafficeast(i, length);
	  #pass_speed_east(cur+j-1, 1) = trafficeast(i, velocity);
    else
      pass_length_east(cur+j-1, 2) = -trafficeast(i, length);
	  #pass_speed_east(cur+j-1, 2) = -trafficeast(i, velocity);
    endif
  endfor
endfor


---------

west2= csvread("west.csv");
east2= csvread("east.csv");


framenum=0:359999;
temp=(framenum/200)';
tempclean= temp;

temp= tempclean;
temp(360000, 5) =0;

for i=1:size(west2,1)
	framenum= west2(i, 1)
	temp(framenum, 2)= west2(i, 2); # left car speed [kph, westward]
	temp(framenum, 3)= west2(i, 3); # left car length [m, westward]
	temp(framenum, 4)= west2(i, 4); # right car speed [kphm, westward]
	temp(framenum, 5)= west2(i, 5); # right car length [m, westward]
endfor
westt=temp;
csvwrite("west2.csv", temp);


--------

the fan graph

# speed
plot( eastt(:,1), eastt(:,2) );
hold on
plot( eastt(:,1), eastt(:,4) );
hold off
set(gca, 'xtick', 0:60:1800);
set(gca, 'ytick', -100:10:100);
set(gca, "linewidth", 2, "fontsize", 4)
axis("tight");
xlabel("Time [s]");
ylabel("Passing vehicle speed [kph]");
title ("Hinotsume Bridge, east span");
filename=sprintf("hinotsume-east-speed.png")
print(filename, '-dpng', '-S2048,480')
close

# length
plot( eastt(:,1), eastt(:,3) );
hold on
plot( eastt(:,1), eastt(:,5) );
hold off
set(gca, 'xtick', 0:60:1800);
set(gca, 'ytick', -24:2:24);
set(gca, "linewidth", 2, "fontsize", 4)
axis("tight");
xlabel("Time [s]");
ylabel("Passing vehicle length [m]");
title ("Hinotsume Bridge, east span");
filename=sprintf("hinotsume-east-length.png")
print(filename, '-dpng', '-S2048,480')
close

# length histogram
clear eastlength;
eastlength= east2(:,3) + east2(:,5);
hist( abs(eastlength), 8)
set(gca, "linewidth", 2, "fontsize", 4)
#axis("tight");
xlabel("Vehicle length [m]");
ylabel("Count [n]");
title ("Hinotsume Bridge, east span");
filename=sprintf("hinotsume-east-lengthhist.png")
print(filename, '-dpng', '-r300')
close

